pkgs/development/libraries/science/math/cudnn/generic.nix

```diff
@@ -74,27 +72,49 @@ in
     #
     # Note also that version <=8.3.0 contained a subdirectory "lib64/" but in
     # version 8.3.2 it seems to have been renamed to simply "lib/".
+    #
+    # doc and dev have special output handling. Other outputs need to be moved to their own
+    # output.
+    # Note that moveToOutput operates on all outputs:
+    # https://github.com/NixOS/nixpkgs/blob/2920b6fc16a9ed5d51429e94238b28306ceda79e/pkgs/build-support/setup-hooks/multiple-outputs.sh#L105-L107
     installPhase =
       ''
         runHook preInstall
 
-        mkdir -p $out
-        cp -a include $out/include
-        [ -d "lib/" ] && cp -a lib $out/lib
-        [ -d "lib64/" ] && cp -a lib64 $out/lib64
-      ''
-      + strings.optionalString removeStatic ''
-        rm -f $out/lib/*.a
-        rm -f $out/lib64/*.a
-      ''
-      + ''
+        mkdir -p "$out"
+        mv * "$out"
+        moveToOutput "lib64" "$lib"
+        moveToOutput "lib" "$lib"
+        moveToOutput "**/*.a" "$static"
```

Comment:
Sanity check: are there no `*.a` archives in `lib` and `lib64`?

Reply:
Oh you're gonna love this -- `moveToOutput` operates on *all* outputs: https://github.com/NixOS/nixpkgs/blob/606c7de99fdc002154e87af16d93a43ec3381c33/pkgs/build-support/setup-hooks/multiple-outputs.sh#L105

So even if we move all the libraries to `lib`, we then move all static libraries to the static `output`.